aten/src/ATen/core/op_registration/op_registration.cpp

```diff
@@ -165,4 +165,57 @@ RegisterOperators::~RegisterOperators() = default;
 RegisterOperators::RegisterOperators(RegisterOperators&&) noexcept = default;
 RegisterOperators& RegisterOperators::operator=(RegisterOperators&&) noexcept = default;
 
+
+CppFunction::CppFunction(KernelFunction func, std::unique_ptr<c10::FunctionSchema> schema)
+  : func_(std::move(func))
+  , schema_(std::move(schema))
+  {}
+
+Module::Module(const char* ns)
+  : ns_(ns)
+  {}
+
+Module::Module(Module&&) noexcept = default;
+Module& Module::operator=(Module&&) noexcept = default;
+
+// TODO: Error if an operator is def'ed multiple times.  Right now we just
+// merge everything
+
+namespace {
+  std::string addNamespace(const char* ns, const char* unqual_name_or_schema) { if (ns) {
+      // TODO: slow!  Fix internal data structures so I don't have to paste the
+      // names together
+      std::ostringstream oss;
+      oss << ns << "::" << unqual_name_or_schema;
```

Comment:
FWIW if we're not actually going to insist that `unqual_name_or_schema` be unqualified we should probably rename it (here and elsewhere). 

Also in that case there's namespace-chaining behavior that we might want to describe in the "New style API" usage comment.

Reply:
BTW curious why this is a function than a private method that just adds `ns_` directly...

Reply:
I guess I can just call it `name_or_schema`. As for why it's a function, it's so that I don't have to declare it as a method in the header file. That's a big win, at least in my book :)